In [284]:
import pandas as pd
import numpy as np

In [285]:
names_col = []
with open('статистика/data.txt', 'r') as file:
    
    for line in file:
        names_col.append(line.strip())

In [286]:
dataframe = pd.read_csv('статистика/horse_data.csv', names=names_col)
dataframe

surgery?  Age  Hospital Number rectal temperature pulse respiratory rate  \
0          2    1           530101              38.50    66               28   
1          1    1           534817               39.2    88               20   
2          2    1           530334              38.30    40               24   
3          1    9          5290409              39.10   164               84   
4          2    1           530255              37.30   104               35   
..       ...  ...              ...                ...   ...              ...   
295        1    1           533886                  ?   120               70   
296        2    1           527702              37.20    72               24   
297        1    1           529386              37.50    72               30   
298        1    1           530612              36.50   100               24   
299        1    1           534618               37.2    40               20   

    temperature of extremities peripheral pulse mucous membranes  \
0                            3                3                ?   
1                            ?                ?                4   
2                            1                1                3   
3                            4                1                6   
4                            ?                ?                6   
..                         ...              ...              ...   
295                          4                ?                4   
296                          3                2                4   
297                          4                3                4   
298                          3                3                3   
299                          ?                ?                ?   

    capillary refill time  ... packed cell volume total protein  \
0                       2  ...              45.00          8.40   
1                       1  ...                 50            85   
2                       1  ...              33.00          6.70   
3                       2  ...              48.00          7.20   
4                       2  ...              74.00          7.40   
..                    ...  ...                ...           ...   
295                     2  ...                 55            65   
296                     2  ...              44.00             ?   
297                     1  ...              60.00          6.80   
298                     1  ...              50.00          6.00   
299                     ?  ...                 36            62   

    abdominocentesis appearance abdomcentesis total protein outcome  \
0                             ?                           ?       2   
1                             2                           2       3   
2                             ?                           ?       1   
3                             3                        5.30       2   
4                             ?                           ?       2   
..                          ...                         ...     ...   
295                           ?                           ?       3   
296                           3                        3.30       3   
297                           ?                           ?       2   
298                           3                        3.40       1   
299                           1                           1       3   

    surgical lesion? type of lesion 1 type of lesion 2 type of lesion 3  \
0                  2            11300                0                0   
1                  2             2208                0                0   
2                  2                0                0                0   
3                  1             2208                0                0   
4                  2             4300                0                0   
..               ...              ...              ...              ...   
295                2             3205      

In [287]:
data_info_raw = dataframe[['surgery?', 'Age', 'rectal temperature', 'pulse', 'respiratory rate', 'temperature of extremities', 'pain', 'outcome']]
data_info_raw

surgery?  Age rectal temperature pulse respiratory rate  \
0          2    1              38.50    66               28   
1          1    1               39.2    88               20   
2          2    1              38.30    40               24   
3          1    9              39.10   164               84   
4          2    1              37.30   104               35   
..       ...  ...                ...   ...              ...   
295        1    1                  ?   120               70   
296        2    1              37.20    72               24   
297        1    1              37.50    72               30   
298        1    1              36.50   100               24   
299        1    1               37.2    40               20   

    temperature of extremities pain outcome  
0                            3    5       2  
1                            ?    3       3  
2                            1    3       1  
3                            4    2       2  
4                            ?    ?       2  
..                         ...  ...     ...  
295                          4    2       3  
296                          3    4       3  
297                          4    4       2  
298                          3    3       1  
299                          ?    ?       3  

[300 rows x 8 columns]

In [288]:
data_info_fill_nan = data_info_raw.replace('?', None)

In [289]:
data_info_fill_nan.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 300 entries, 0 to 299
Data columns (total 8 columns):
 #   Column                      Non-Null Count  Dtype 
---  ------                      --------------  ----- 
 0   surgery?                    299 non-null    object
 1   Age                         300 non-null    int64 
 2   rectal temperature          240 non-null    object
 3   pulse                       276 non-null    object
 4   respiratory rate            242 non-null    object
 5   temperature of extremities  244 non-null    object
 6   pain                        245 non-null    object
 7   outcome                     299 non-null    object
dtypes: int64(1), object(7)
memory usage: 18.9+ KB


В датафрейме пропущено менее 20% значений, что в принципе не так страшно, при этом также стоит отметить, что строк с пропущенными значениями также не так много, что позволит нам не обращать внимание на пропуски в целом

In [290]:
data_info_float = data_info_fill_nan.astype({'surgery?': np.float64, 'rectal temperature': np.float64, 'pulse': np.float64, 'respiratory rate': np.float64, 'temperature of extremities': np.float64, 'pain': np.float64, 'outcome': np.float64})

In [291]:
data_info_float.describe()

surgery?         Age  rectal temperature       pulse  \
count  299.000000  300.000000          240.000000  276.000000   
mean     1.397993    1.640000           38.167917   71.913043   
std      0.490305    2.173972            0.732289   28.630557   
min      1.000000    1.000000           35.400000   30.000000   
25%      1.000000    1.000000           37.800000   48.000000   
50%      1.000000    1.000000           38.200000   64.000000   
75%      2.000000    1.000000           38.500000   88.000000   
max      2.000000    9.000000           40.800000  184.000000   

       respiratory rate  temperature of extremities       pain     outcome  
count        242.000000                  244.000000  245.00000  299.000000  
mean          30.417355                    2.348361    2.95102    1.551839  
std           17.642231                    1.045054    1.30794    0.737187  
min            8.000000                    1.000000    1.00000    1.000000  
25%           18.500000                    1.000000    2.00000    1.000000  
50%           24.500000                    3.000000    3.00000    1.000000  
75%           36.000000                    3.000000    4.00000    2.000000  
max           96.000000                    4.000000    5.00000    3.000000

In [292]:
(data_info_float.isna().mean() * 100).round(2)

surgery?                       0.33
Age                            0.00
rectal temperature            20.00
pulse                          8.00
respiratory rate              19.33
temperature of extremities    18.67
pain                          18.33
outcome                        0.33
dtype: float64

# Взрослые лошади

Для начала рассмотрим статистику по взрослым лошадям:

## Смертность прооперированных лошадей 

In [293]:
data_info_adult_surg = data_info_float[(data_info_float['Age'] == 1) & (data_info_float['surgery?'] == 1)]
data_info_adult_surg

surgery?  Age  rectal temperature  pulse  respiratory rate  \
1         1.0    1                39.2   88.0              20.0   
6         1.0    1                37.9   48.0              16.0   
7         1.0    1                 NaN   60.0               NaN   
10        1.0    1                38.1   66.0              12.0   
12        1.0    1                37.2   42.0              12.0   
..        ...  ...                 ...    ...               ...   
293       1.0    1                 NaN   78.0              24.0   
295       1.0    1                 NaN  120.0              70.0   
297       1.0    1                37.5   72.0              30.0   
298       1.0    1                36.5  100.0              24.0   
299       1.0    1                37.2   40.0              20.0   

     temperature of extremities  pain  outcome  
1                           NaN   3.0      3.0  
6                           1.0   3.0      1.0  
7                           3.0   NaN      2.0  
10                          3.0   3.0      1.0  
12                          2.0   3.0      1.0  
..                          ...   ...      ...  
293                         3.0   NaN      3.0  
295                         4.0   2.0      3.0  
297                         4.0   4.0      2.0  
298                         3.0   3.0      1.0  
299                         NaN   NaN      3.0  

[162 rows x 8 columns]

In [294]:
data_info_adult_surg['outcome'].value_counts()

1.0    88
2.0    48
3.0    26
Name: outcome, dtype: int64

Смертность взрослых лошадей после проведения операции составляет практически половину: 45,7%

Сначала уберем выпадающие значения по пульсу

In [310]:
q1 = data_info_adult_surg['pulse'].quantile(0.25)
q3 = data_info_adult_surg['pulse'].quantile(0.75)
iqr = q3 - q1
lower_bound = q1 - (1.5 * iqr) 
upper_bound = q3 + (1.5 * iqr)
data_info_as = data_info_adult_surg[data_info_adult_surg['pulse'].between(lower_bound, upper_bound, inclusive=True)]
data_info_as

/var/folders/xs/yywhsxs13dnc369zl2gzp53m0000gn/T/ipykernel_4514/646406692.py:6: FutureWarning: Boolean inputs to the `inclusive` argument are deprecated in favour of `both` or `neither`.
  data_info_as = data_info_adult_surg[data_info_adult_surg['pulse'].between(lower_bound, upper_bound, inclusive=True)]


surgery?  Age  rectal temperature  pulse  respiratory rate  \
1         1.0    1                39.2   88.0              20.0   
6         1.0    1                37.9   48.0              16.0   
7         1.0    1                 NaN   60.0               NaN   
10        1.0    1                38.1   66.0              12.0   
12        1.0    1                37.2   42.0              12.0   
..        ...  ...                 ...    ...               ...   
293       1.0    1                 NaN   78.0              24.0   
295       1.0    1                 NaN  120.0              70.0   
297       1.0    1                37.5   72.0              30.0   
298       1.0    1                36.5  100.0              24.0   
299       1.0    1                37.2   40.0              20.0   

     temperature of extremities  pain  outcome  
1                           NaN   3.0      3.0  
6                           1.0   3.0      1.0  
7                           3.0   NaN      2.0  
10                          3.0   3.0      1.0  
12                          2.0   3.0      1.0  
..                          ...   ...      ...  
293                         3.0   NaN      3.0  
295                         4.0   2.0      3.0  
297                         4.0   4.0      2.0  
298                         3.0   3.0      1.0  
299                         NaN   NaN      3.0  

[148 rows x 8 columns]

Теперь отсортируем значения data_info_adult_surg по значению 2 - погибли и затем 3 - эвтаназия
Далее посмотрим статистику по всем остальным значениям таблицы:

### Естесственная смерть

In [311]:
data_info_asd = data_info_as[data_info_as['outcome'] == 2]
data_info_asd['pain'].value_counts()

4.0    12
5.0    10
2.0     9
3.0     3
Name: pain, dtype: int64

Исходя из описания датафрейма видим, что смертность лошади обуславливается одним из ключевых показателей: температура, пульс, частота дыхания и субъективный показатель - боль. 

Чаще всего лошади умирали испытывая боль (значения 4 - периодическая сильная боль, 5 - постоянная сильная боль) в редких случаях боль была либо 2 - ноющая, либо переодически средняя. 

Пропуски температуры в данном случае вполне рандомны, то есть лошадь могла испытывать сильные боли при низкой температуре и наоборот. Заполнить их можно средним значением по сегменту. 

In [322]:
data_info_asd['rectal temperature'] = data_info_asd['rectal temperature'].fillna(data_info_asd['rectal temperature'].mean())

/var/folders/xs/yywhsxs13dnc369zl2gzp53m0000gn/T/ipykernel_4514/2542292129.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_info_asd['rectal temperature'] = data_info_asd['rectal temperature'].fillna(data_info_asd['rectal temperature'].mean())


In [313]:
data_info_asd

surgery?  Age  rectal temperature  pulse  respiratory rate  \
7         1.0    1           38.007143   60.0               NaN   
15        1.0    1           37.600000   96.0              48.0   
40        1.0    1           38.007143   88.0               NaN   
48        1.0    1           38.900000   80.0              44.0   
51        1.0    1           37.400000   84.0              36.0   
57        1.0    1           38.007143   64.0              36.0   
62        1.0    1           37.800000   88.0              22.0   
72        1.0    1           37.700000   56.0               NaN   
73        1.0    1           38.100000   52.0              24.0   
80        1.0    1           36.400000   98.0              35.0   
86        1.0    1           38.007143   88.0              40.0   
104       1.0    1           37.800000   52.0              24.0   
110       1.0    1           38.007143   70.0              16.0   
113       1.0    1           36.800000   60.0              28.0   
123       1.0    1           38.200000   64.0              24.0   
130       1.0    1           37.800000   30.0              12.0   
134       1.0    1           38.007143  100.0               NaN   
142       1.0    1           38.007143  120.0               NaN   
161       1.0    1           38.300000   42.0              24.0   
173       1.0    1           38.007143  104.0               NaN   
177       1.0    1           38.007143   68.0              14.0   
180       1.0    1           38.400000   84.0              30.0   
193       1.0    1           38.007143   60.0              48.0   
201       1.0    1           38.500000   86.0               NaN   
210       1.0    1           37.900000   68.0              20.0   
211       1.0    1           38.000000   86.0              24.0   
226       1.0    1           39.200000  120.0              20.0   
235       1.0    1           38.100000   52.0              24.0   
240       1.0    1           38.000000   90.0              30.0   
242       1.0    1           38.007143   36.0              32.0   
247       1.0    1           38.100000   88.0              24.0   
257       1.0    1           38.700000   96.0              28.0   
259       1.0    1           40.800000   72.0              42.0   
263       1.0    1           37.100000   75.0              36.0   
270       1.0    1           37.500000   60.0              50.0   
273       1.0    1           37.700000  120.0              28.0   
282       1.0    1           38.007143   70.0              16.0   
291       1.0    1           38.900000   80.0              44.0   
292       1.0    1           37.000000   66.0              20.0   
297       1.0    1           37.500000   72.0              30.0   

     temperature of extremities  pain  outcome  
7                           3.0   NaN      2.0  
15                          3.0   5.0      2.0  
40                          3.0   5.0      2.0  
48                          3.0   2.0      2.0  
51                          1.0   3.0      2.0  
57                          2.0   5.0      2.0  
62                          2.0   3.0      2.0  
72                          NaN   NaN      2.0  
73                          1.0   4.0      2.0  
80                          3.0   4.0      2.0  
86                          3.0   5.0      2.0  
104                         1.0   4.0      2.0  
110                         3.0   2.0      2.0  
113                         NaN   NaN      2.0  
123                         1.0   4.0      2.0  
130                         NaN   NaN      2.0  
134                         3.0   5.0      2.0  
142                         4.0   5.0      2.0  
161                         NaN   NaN      2.0  
173                         3.0   2.0      2.0  
177                         NaN   4.0      2.0  
180                         3.0   4.0      2.0  
193                         3.0   4.0      2.0  
201                         1.0   4.0      2.0  
210                         NaN 

С учетом того, что частота дыхания для лошадей в этой категории выше нормального либо значительно, либо средне, то вполне можно пропуски заменить на модальное значение.  
Точно так же заполним значения по боли (хотя помним, что это субъективный показатель), а также значения статистики, посчитаной ранее

In [314]:
data_info_asd['respiratory rate'] = data_info_asd['respiratory rate'].fillna(data_info_asd['respiratory rate'].mode()[0])
data_info_asd['pain'] = data_info_asd['pain'].fillna(data_info_asd['pain'].mode()[0])
data_info_asd

/var/folders/xs/yywhsxs13dnc369zl2gzp53m0000gn/T/ipykernel_4514/1278162081.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_info_asd['respiratory rate'] = data_info_asd['respiratory rate'].fillna(data_info_asd['respiratory rate'].mode()[0])
/var/folders/xs/yywhsxs13dnc369zl2gzp53m0000gn/T/ipykernel_4514/1278162081.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_info_asd['pain'] = data_info_asd['pain'].fillna(data_info_asd['pain'].mode()[0])


surgery?  Age  rectal temperature  pulse  respiratory rate  \
7         1.0    1           38.007143   60.0              24.0   
15        1.0    1           37.600000   96.0              48.0   
40        1.0    1           38.007143   88.0              24.0   
48        1.0    1           38.900000   80.0              44.0   
51        1.0    1           37.400000   84.0              36.0   
57        1.0    1           38.007143   64.0              36.0   
62        1.0    1           37.800000   88.0              22.0   
72        1.0    1           37.700000   56.0              24.0   
73        1.0    1           38.100000   52.0              24.0   
80        1.0    1           36.400000   98.0              35.0   
86        1.0    1           38.007143   88.0              40.0   
104       1.0    1           37.800000   52.0              24.0   
110       1.0    1           38.007143   70.0              16.0   
113       1.0    1           36.800000   60.0              28.0   
123       1.0    1           38.200000   64.0              24.0   
130       1.0    1           37.800000   30.0              12.0   
134       1.0    1           38.007143  100.0              24.0   
142       1.0    1           38.007143  120.0              24.0   
161       1.0    1           38.300000   42.0              24.0   
173       1.0    1           38.007143  104.0              24.0   
177       1.0    1           38.007143   68.0              14.0   
180       1.0    1           38.400000   84.0              30.0   
193       1.0    1           38.007143   60.0              48.0   
201       1.0    1           38.500000   86.0              24.0   
210       1.0    1           37.900000   68.0              20.0   
211       1.0    1           38.000000   86.0              24.0   
226       1.0    1           39.200000  120.0              20.0   
235       1.0    1           38.100000   52.0              24.0   
240       1.0    1           38.000000   90.0              30.0   
242       1.0    1           38.007143   36.0              32.0   
247       1.0    1           38.100000   88.0              24.0   
257       1.0    1           38.700000   96.0              28.0   
259       1.0    1           40.800000   72.0              42.0   
263       1.0    1           37.100000   75.0              36.0   
270       1.0    1           37.500000   60.0              50.0   
273       1.0    1           37.700000  120.0              28.0   
282       1.0    1           38.007143   70.0              16.0   
291       1.0    1           38.900000   80.0              44.0   
292       1.0    1           37.000000   66.0              20.0   
297       1.0    1           37.500000   72.0              30.0   

     temperature of extremities  pain  outcome  
7                           3.0   4.0      2.0  
15                          3.0   5.0      2.0  
40                          3.0   5.0      2.0  
48                          3.0   2.0      2.0  
51                          1.0   3.0      2.0  
57                          2.0   5.0      2.0  
62                          2.0   3.0      2.0  
72                          NaN   4.0      2.0  
73                          1.0   4.0      2.0  
80                          3.0   4.0      2.0  
86                          3.0   5.0      2.0  
104                         1.0   4.0      2.0  
110                         3.0   2.0      2.0  
113                         NaN   4.0      2.0  
123                         1.0   4.0      2.0  
130                         NaN   4.0      2.0  
134                         3.0   5.0      2.0  
142                         4.0   5.0      2.0  
161                         NaN   4.0      2.0  
173                         3.0   2.0      2.0  
177                         NaN   4.0      2.0  
180                         3.0   4.0      2.0  
193                         3.0   4.0      2.0  
201                         1.0   4.0      2.0  
210                         NaN 

С пульсом все сложнее, поэтому решил заменить на медиану
Экстримы температуры заполнил медианой

In [316]:
data_info_asd['pulse'] = data_info_asd['pulse'].fillna(data_info_asd['pulse'].median())
data_info_asd['temperature of extremities'] = data_info_asd['temperature of extremities'].fillna(data_info_asd['temperature of extremities'].median())
data_info_asd

/var/folders/xs/yywhsxs13dnc369zl2gzp53m0000gn/T/ipykernel_4514/3142639789.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_info_asd['pulse'] = data_info_asd['pulse'].fillna(data_info_asd['pulse'].median())
/var/folders/xs/yywhsxs13dnc369zl2gzp53m0000gn/T/ipykernel_4514/3142639789.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_info_asd['temperature of extremities'] = data_info_asd['temperature of extremities'].fillna(data_info_asd['temperature of extremities'].median())


surgery?  Age  rectal temperature  pulse  respiratory rate  \
7         1.0    1           38.007143   60.0              24.0   
15        1.0    1           37.600000   96.0              48.0   
40        1.0    1           38.007143   88.0              24.0   
48        1.0    1           38.900000   80.0              44.0   
51        1.0    1           37.400000   84.0              36.0   
57        1.0    1           38.007143   64.0              36.0   
62        1.0    1           37.800000   88.0              22.0   
72        1.0    1           37.700000   56.0              24.0   
73        1.0    1           38.100000   52.0              24.0   
80        1.0    1           36.400000   98.0              35.0   
86        1.0    1           38.007143   88.0              40.0   
104       1.0    1           37.800000   52.0              24.0   
110       1.0    1           38.007143   70.0              16.0   
113       1.0    1           36.800000   60.0              28.0   
123       1.0    1           38.200000   64.0              24.0   
130       1.0    1           37.800000   30.0              12.0   
134       1.0    1           38.007143  100.0              24.0   
142       1.0    1           38.007143  120.0              24.0   
161       1.0    1           38.300000   42.0              24.0   
173       1.0    1           38.007143  104.0              24.0   
177       1.0    1           38.007143   68.0              14.0   
180       1.0    1           38.400000   84.0              30.0   
193       1.0    1           38.007143   60.0              48.0   
201       1.0    1           38.500000   86.0              24.0   
210       1.0    1           37.900000   68.0              20.0   
211       1.0    1           38.000000   86.0              24.0   
226       1.0    1           39.200000  120.0              20.0   
235       1.0    1           38.100000   52.0              24.0   
240       1.0    1           38.000000   90.0              30.0   
242       1.0    1           38.007143   36.0              32.0   
247       1.0    1           38.100000   88.0              24.0   
257       1.0    1           38.700000   96.0              28.0   
259       1.0    1           40.800000   72.0              42.0   
263       1.0    1           37.100000   75.0              36.0   
270       1.0    1           37.500000   60.0              50.0   
273       1.0    1           37.700000  120.0              28.0   
282       1.0    1           38.007143   70.0              16.0   
291       1.0    1           38.900000   80.0              44.0   
292       1.0    1           37.000000   66.0              20.0   
297       1.0    1           37.500000   72.0              30.0   

     temperature of extremities  pain  outcome  
7                           3.0   4.0      2.0  
15                          3.0   5.0      2.0  
40                          3.0   5.0      2.0  
48                          3.0   2.0      2.0  
51                          1.0   3.0      2.0  
57                          2.0   5.0      2.0  
62                          2.0   3.0      2.0  
72                          3.0   4.0      2.0  
73                          1.0   4.0      2.0  
80                          3.0   4.0      2.0  
86                          3.0   5.0      2.0  
104                         1.0   4.0      2.0  
110                         3.0   2.0      2.0  
113                         3.0   4.0      2.0  
123                         1.0   4.0      2.0  
130                         3.0   4.0      2.0  
134                         3.0   5.0      2.0  
142                         4.0   5.0      2.0  
161                         3.0   4.0      2.0  
173                         3.0   2.0      2.0  
177                         3.0   4.0      2.0  
180                         3.0   4.0      2.0  
193                         3.0   4.0      2.0  
201                         1.0   4.0      2.0  
210                         3.0 

In [326]:
info_asd = data_info_asd.describe()
info_asd.round()

surgery?   Age  rectal temperature  pulse  respiratory rate  \
count      40.0  40.0                40.0   40.0              40.0   
mean        1.0   1.0                38.0   76.0              29.0   
std         0.0   0.0                 1.0   22.0              10.0   
min         1.0   1.0                36.0   30.0              12.0   
25%         1.0   1.0                38.0   60.0              24.0   
50%         1.0   1.0                38.0   74.0              24.0   
75%         1.0   1.0                38.0   88.0              35.0   
max         1.0   1.0                41.0  120.0              50.0   

       temperature of extremities  pain  outcome  
count                        40.0  40.0     40.0  
mean                          3.0   4.0      2.0  
std                           1.0   1.0      0.0  
min                           1.0   2.0      2.0  
25%                           3.0   3.0      2.0  
50%                           3.0   4.0      2.0  
75%                           3.0   4.0      2.0  
max                           4.0   5.0      2.0

## Итого по взрослым лошадям с проведенной операцией и естесственной смертью

### Нормальная температура взрослых лошадей 37,8 гр С

Средняя температура 38 гр С  
СКО 1 гр С  
Минимальная температура 36 гр С  
Максимальная температура 41 гр С  

С учетом того, что 75% также 38 гр С, то можно сделать заключение, что средняя температура не сильно превышает нормальную. 

### Нормальный пульс 30 - 40 уд/мин, но возможно и 20-30

Средний пульс 76 уд/мин  
СКО 22 уд/мин  
Минимальный пульс 30 уд/мин  
Максимальный пульс 120 уд/мин  

Средний пульс значительно превышает нормальное значение, начиная с первого квартиля также идет превышение значения

### Нормальная частота дыхания 8 - 10

Средняя частота дыхания 29  
СКО 10  
Минимальная частота дыхания 12  
Максимальная 50  

Среднее значение частоты дыхания почти в 3 раза превышает нормальное значение.  

### Боль

В основном лошади испытывали сильную боль либо переодически либо постоянно. 


## Эвтаназия

In [320]:
data_info_ase = data_info_as[data_info_as['outcome'] == 3]
data_info_ase['pain'].value_counts()

2.0    7
5.0    6
4.0    5
3.0    1
Name: pain, dtype: int64

Замена значений проведем аналогичным методом что и для естесственной смерти

In [323]:
data_info_ase['rectal temperature'] = data_info_ase['rectal temperature'].fillna(data_info_ase['rectal temperature'].mean())
data_info_ase['respiratory rate'] = data_info_ase['respiratory rate'].fillna(data_info_ase['respiratory rate'].mode()[0])
data_info_ase['pain'] = data_info_ase['pain'].fillna(data_info_ase['pain'].mode()[0])
data_info_ase['pulse'] = data_info_ase['pulse'].fillna(data_info_ase['pulse'].median())
data_info_ase['temperature of extremities'] = data_info_ase['temperature of extremities'].fillna(data_info_ase['temperature of extremities'].median())
data_info_ase

/var/folders/xs/yywhsxs13dnc369zl2gzp53m0000gn/T/ipykernel_4514/2932916932.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_info_ase['rectal temperature'] = data_info_ase['rectal temperature'].fillna(data_info_ase['rectal temperature'].mean())
/var/folders/xs/yywhsxs13dnc369zl2gzp53m0000gn/T/ipykernel_4514/2932916932.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_info_ase['respiratory rate'] = data_info_ase['respiratory rate'].fillna(data_info_ase['respiratory rate'].mode()[0])
/var/folder

surgery?  Age  rectal temperature  pulse  respiratory rate  \
1         1.0    1               39.20   88.0              20.0   
43        1.0    1               37.97  120.0              20.0   
44        1.0    1               35.40  140.0              24.0   
97        1.0    1               38.10   88.0              32.0   
101       1.0    1               37.97   60.0              20.0   
108       1.0    1               38.70   60.0              32.0   
121       1.0    1               38.30   80.0              40.0   
128       1.0    1               37.40   84.0              20.0   
144       1.0    1               37.10   84.0              40.0   
157       1.0    1               37.80   60.0              24.0   
163       1.0    1               38.00   66.0              20.0   
165       1.0    1               39.40  120.0              48.0   
169       1.0    1               37.97   65.0              24.0   
202       1.0    1               37.50   48.0              40.0   
208       1.0    1               37.80   88.0              80.0   
218       1.0    1               38.10   40.0              36.0   
220       1.0    1               39.20   88.0              58.0   
228       1.0    1               37.30   90.0              40.0   
239       1.0    1               38.10   70.0              22.0   
250       1.0    1               39.30  100.0              51.0   
256       1.0    1               37.50   72.0              20.0   
274       1.0    1               37.97   76.0              20.0   
276       1.0    1               38.00   36.0              16.0   
293       1.0    1               37.97   78.0              24.0   
295       1.0    1               37.97  120.0              70.0   
299       1.0    1               37.20   40.0              20.0   

     temperature of extremities  pain  outcome  
1                           3.0   3.0      3.0  
43                          3.0   4.0      3.0  
44                          3.0   4.0      3.0  
97                          3.0   2.0      3.0  
101                         4.0   5.0      3.0  
108                         4.0   4.0      3.0  
121                         3.0   4.0      3.0  
128                         3.0   2.0      3.0  
144                         3.0   2.0      3.0  
157                         1.0   2.0      3.0  
163                         1.0   5.0      3.0  
165                         3.0   2.0      3.0  
169                         3.0   5.0      3.0  
202                         3.0   2.0      3.0  
208                         3.0   2.0      3.0  
218                         1.0   2.0      3.0  
220                         4.0   5.0      3.0  
228                         3.0   5.0      3.0  
239                         3.0   5.0      3.0  
250                         4.0   2.0      3.0  
256                         2.0   2.0      3.0  
274                         3.0   2.0      3.0  
276                         3.0   4.0      3.0  
293                         3.0   2.0      3.0  
295                         4.0   2.0      3.0  
299                         3.0   2.0      3.0

In [328]:
info_ase = data_info_ase.describe()
info_ase.round()

surgery?   Age  rectal temperature  pulse  respiratory rate  \
count      26.0  26.0                26.0   26.0              26.0   
mean        1.0   1.0                38.0   79.0              33.0   
std         0.0   0.0                 1.0   26.0              17.0   
min         1.0   1.0                35.0   36.0              16.0   
25%         1.0   1.0                38.0   61.0              20.0   
50%         1.0   1.0                38.0   79.0              24.0   
75%         1.0   1.0                38.0   88.0              40.0   
max         1.0   1.0                39.0  140.0              80.0   

       temperature of extremities  pain  outcome  
count                        26.0  26.0     26.0  
mean                          3.0   3.0      3.0  
std                           1.0   1.0      0.0  
min                           1.0   2.0      3.0  
25%                           3.0   2.0      3.0  
50%                           3.0   2.0      3.0  
75%                           3.0   4.0      3.0  
max                           4.0   5.0      3.0

## Итого по взрослым лошадям с проведенной операцией и эвтаназией

### Нормальная температура взрослых лошадей 37,8 гр С

Средняя температура 38 гр С  
СКО 1 гр С  
Минимальная температура 35 гр С  
Максимальная температура 39 гр С  

С учетом того, что 75% также 38 гр С, то можно сделать заключение, что средняя температура не сильно превышает нормальную. 

### Нормальный пульс 30 - 40 уд/мин, но возможно и 20-30

Средний пульс 79 уд/мин  
СКО 26 уд/мин  
Минимальный пульс 36 уд/мин  
Максимальный пульс 140 уд/мин  

Средний пульс значительно превышает нормальное значение, начиная с первого квартиля также идет превышение значения

### Нормальная частота дыхания 8 - 10

Средняя частота дыхания 33  
СКО 17  
Минимальная частота дыхания 16  
Максимальная 80  

Среднее значение частоты дыхания почти в 3 раза превышает нормальное значение.  

### Боль

В основном лошади испытывали боль как сильную так и среднюю равномерно. 


## Выжившие лошади

In [330]:
data_info_asl = data_info_as[data_info_as['outcome'] == 1]
data_info_asl['pain'].value_counts()

3.0    30
2.0    13
5.0     9
1.0     9
4.0     6
Name: pain, dtype: int64

In [332]:
data_info_asl['rectal temperature'] = data_info_asl['rectal temperature'].fillna(data_info_asl['rectal temperature'].mean())
data_info_asl['respiratory rate'] = data_info_asl['respiratory rate'].fillna(data_info_asl['respiratory rate'].mode()[0])
data_info_asl['pain'] = data_info_asl['pain'].fillna(data_info_asl['pain'].mode()[0])
data_info_asl['pulse'] = data_info_asl['pulse'].fillna(data_info_asl['pulse'].median())
data_info_asl['temperature of extremities'] = data_info_asl['temperature of extremities'].fillna(data_info_asl['temperature of extremities'].median())
data_info_asl

/var/folders/xs/yywhsxs13dnc369zl2gzp53m0000gn/T/ipykernel_4514/1019345346.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_info_asl['rectal temperature'] = data_info_asl['rectal temperature'].fillna(data_info_asl['rectal temperature'].mean())
/var/folders/xs/yywhsxs13dnc369zl2gzp53m0000gn/T/ipykernel_4514/1019345346.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_info_asl['respiratory rate'] = data_info_asl['respiratory rate'].fillna(data_info_asl['respiratory rate'].mode()[0])
/var/folder

surgery?  Age  rectal temperature  pulse  respiratory rate  \
6         1.0    1                37.9   48.0              16.0   
10        1.0    1                38.1   66.0              12.0   
12        1.0    1                37.2   42.0              12.0   
14        1.0    1                38.2   76.0              28.0   
18        1.0    1                37.6   64.0              21.0   
..        ...  ...                 ...    ...               ...   
280       1.0    1                38.6   48.0              12.0   
283       1.0    1                38.2   72.0              18.0   
285       1.0    1                38.5   66.0              24.0   
289       1.0    1                38.0   50.0              36.0   
298       1.0    1                36.5  100.0              24.0   

     temperature of extremities  pain  outcome  
6                           1.0   3.0      1.0  
10                          3.0   3.0      1.0  
12                          2.0   3.0      1.0  
14                          3.0   3.0      1.0  
18                          1.0   2.0      1.0  
..                          ...   ...      ...  
280                         2.0   1.0      1.0  
283                         2.0   3.0      1.0  
285                         1.0   3.0      1.0  
289                         2.0   3.0      1.0  
298                         3.0   3.0      1.0  

[82 rows x 8 columns]

In [333]:
info_asl = data_info_asl.describe()
info_asl.round()

surgery?   Age  rectal temperature  pulse  respiratory rate  \
count      82.0  82.0                82.0   82.0              82.0   
mean        1.0   1.0                38.0   65.0              27.0   
std         0.0   0.0                 1.0   23.0              20.0   
min         1.0   1.0                36.0   30.0               8.0   
25%         1.0   1.0                38.0   48.0              12.0   
50%         1.0   1.0                38.0   60.0              20.0   
75%         1.0   1.0                38.0   76.0              30.0   
max         1.0   1.0                40.0  132.0              96.0   

       temperature of extremities  pain  outcome  
count                        82.0  82.0     82.0  
mean                          2.0   3.0      1.0  
std                           1.0   1.0      0.0  
min                           1.0   1.0      1.0  
25%                           1.0   2.0      1.0  
50%                           2.0   3.0      1.0  
75%                           3.0   3.0      1.0  
max                           4.0   5.0      1.0

## Итого по взрослым лошадям с проведенной операцией и выживших

### Нормальная температура взрослых лошадей 37,8 гр С

Средняя температура 38 гр С  
СКО 1 гр С  
Минимальная температура 36 гр С  
Максимальная температура 40 гр С  

С учетом того, что 75% также 38 гр С, то можно сделать заключение, что средняя температура не сильно превышает нормальную. 

### Нормальный пульс 30 - 40 уд/мин, но возможно и 20-30

Средний пульс 65 уд/мин  
СКО 23 уд/мин  
Минимальный пульс 30 уд/мин  
Максимальный пульс 132 уд/мин  

Средний пульс значительно превышает нормальное значение, начиная с первого квартиля также идет превышение значения

### Нормальная частота дыхания 8 - 10

Средняя частота дыхания 27  
СКО 20  
Минимальная частота дыхания 8  
Максимальная 96  

Среднее значение частоты дыхания почти в 3 раза превышает нормальное значение.  

### Боль

В основном лошади испытывали либо ноющую либо среднюю переодически боль. 


## Смертность лошадей без операции

In [334]:
data_info_adult_nosurg = data_info_float[(data_info_float['Age'] == 1) & (data_info_float['surgery?'] == 2)]
data_info_adult_nosurg

surgery?  Age  rectal temperature  pulse  respiratory rate  \
0         2.0    1                38.5   66.0              28.0   
2         2.0    1                38.3   40.0              24.0   
4         2.0    1                37.3  104.0              35.0   
5         2.0    1                 NaN    NaN               NaN   
8         2.0    1                 NaN   80.0              36.0   
..        ...  ...                 ...    ...               ...   
284       2.0    1                38.5   54.0               NaN   
286       2.0    1                37.8   82.0              12.0   
290       2.0    1                38.6   45.0              16.0   
294       2.0    1                38.5   40.0              16.0   
296       2.0    1                37.2   72.0              24.0   

     temperature of extremities  pain  outcome  
0                           3.0   5.0      2.0  
2                           1.0   3.0      1.0  
4                           NaN   NaN      2.0  
5                           2.0   2.0      1.0  
8                           3.0   4.0      3.0  
..                          ...   ...      ...  
284                         1.0   3.0      1.0  
286                         3.0   4.0      3.0  
290                         2.0   1.0      1.0  
294                         1.0   2.0      1.0  
296                         3.0   4.0      3.0  

[113 rows x 8 columns]

In [335]:
data_info_adult_nosurg['outcome'].value_counts()

1.0    78
2.0    18
3.0    17
Name: outcome, dtype: int64

Смертность лошадей без операционного вмешательства меньше вполовину. 

In [336]:
q1 = data_info_adult_nosurg['pulse'].quantile(0.25)
q3 = data_info_adult_nosurg['pulse'].quantile(0.75)
iqr = q3 - q1
lower_bound = q1 - (1.5 * iqr) 
upper_bound = q3 + (1.5 * iqr)
data_info_an = data_info_adult_nosurg[data_info_adult_nosurg['pulse'].between(lower_bound, upper_bound, inclusive=True)]
data_info_an

/var/folders/xs/yywhsxs13dnc369zl2gzp53m0000gn/T/ipykernel_4514/1090549952.py:6: FutureWarning: Boolean inputs to the `inclusive` argument are deprecated in favour of `both` or `neither`.
  data_info_an = data_info_adult_nosurg[data_info_adult_nosurg['pulse'].between(lower_bound, upper_bound, inclusive=True)]


surgery?  Age  rectal temperature  pulse  respiratory rate  \
0         2.0    1                38.5   66.0              28.0   
2         2.0    1                38.3   40.0              24.0   
4         2.0    1                37.3  104.0              35.0   
8         2.0    1                 NaN   80.0              36.0   
11        2.0    1                39.1   72.0              52.0   
..        ...  ...                 ...    ...               ...   
284       2.0    1                38.5   54.0               NaN   
286       2.0    1                37.8   82.0              12.0   
290       2.0    1                38.6   45.0              16.0   
294       2.0    1                38.5   40.0              16.0   
296       2.0    1                37.2   72.0              24.0   

     temperature of extremities  pain  outcome  
0                           3.0   5.0      2.0  
2                           1.0   3.0      1.0  
4                           NaN   NaN      2.0  
8                           3.0   4.0      3.0  
11                          2.0   2.0      1.0  
..                          ...   ...      ...  
284                         1.0   3.0      1.0  
286                         3.0   4.0      3.0  
290                         2.0   1.0      1.0  
294                         1.0   2.0      1.0  
296                         3.0   4.0      3.0  

[104 rows x 8 columns]

### Естесственная смерть

In [338]:
data_info_and = data_info_an[data_info_an['outcome'] == 2]
data_info_and['pain'].value_counts()

5.0    6
2.0    3
3.0    3
4.0    1
1.0    1
Name: pain, dtype: int64

In [339]:
data_info_and['rectal temperature'] = data_info_and['rectal temperature'].fillna(data_info_and['rectal temperature'].mean())
data_info_and['respiratory rate'] = data_info_and['respiratory rate'].fillna(data_info_and['respiratory rate'].mode()[0])
data_info_and['pain'] = data_info_and['pain'].fillna(data_info_and['pain'].mode()[0])
data_info_and['pulse'] = data_info_and['pulse'].fillna(data_info_and['pulse'].median())
data_info_and['temperature of extremities'] = data_info_and['temperature of extremities'].fillna(data_info_and['temperature of extremities'].median())
data_info_and

/var/folders/xs/yywhsxs13dnc369zl2gzp53m0000gn/T/ipykernel_4514/3602287389.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_info_and['rectal temperature'] = data_info_and['rectal temperature'].fillna(data_info_and['rectal temperature'].mean())
/var/folders/xs/yywhsxs13dnc369zl2gzp53m0000gn/T/ipykernel_4514/3602287389.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_info_and['respiratory rate'] = data_info_and['respiratory rate'].fillna(data_info_and['respiratory rate'].mode()[0])
/var/folder

surgery?  Age  rectal temperature  pulse  respiratory rate  \
0         2.0    1           38.500000   66.0              28.0   
4         2.0    1           37.300000  104.0              35.0   
30        2.0    1           37.700000   96.0              30.0   
45        2.0    1           38.169231  120.0              36.0   
49        2.0    1           37.200000   84.0              48.0   
59        2.0    1           38.169231   96.0              36.0   
91        2.0    1           40.300000  114.0              36.0   
131       2.0    1           37.600000   88.0              36.0   
141       2.0    1           36.000000   42.0              30.0   
147       2.0    1           38.000000   44.0              12.0   
171       2.0    1           39.000000   86.0              16.0   
197       2.0    1           38.500000   96.0              36.0   
231       2.0    1           39.500000   92.0              28.0   
245       2.0    1           38.169231   96.0              36.0   
251       2.0    1           36.600000   42.0              18.0   
281       2.0    1           40.000000   78.0              36.0   

     temperature of extremities  pain  outcome  
0                           3.0   5.0      2.0  
4                           3.0   5.0      2.0  
30                          3.0   5.0      2.0  
45                          4.0   5.0      2.0  
49                          3.0   4.0      2.0  
59                          3.0   5.0      2.0  
91                          3.0   2.0      2.0  
131                         3.0   3.0      2.0  
141                         3.0   5.0      2.0  
147                         2.0   3.0      2.0  
171                         3.0   3.0      2.0  
197                         3.0   2.0      2.0  
231                         3.0   5.0      2.0  
245                         3.0   5.0      2.0  
251                         3.0   1.0      2.0  
281                         3.0   2.0      2.0

In [343]:
info_and = data_info_and.describe()
info_and.round()

surgery?   Age  rectal temperature  pulse  respiratory rate  \
count      16.0  16.0                16.0   16.0              16.0   
mean        2.0   1.0                38.0   84.0              31.0   
std         0.0   0.0                 1.0   24.0               9.0   
min         2.0   1.0                36.0   42.0              12.0   
25%         2.0   1.0                38.0   75.0              28.0   
50%         2.0   1.0                38.0   90.0              36.0   
75%         2.0   1.0                39.0   96.0              36.0   
max         2.0   1.0                40.0  120.0              48.0   

       temperature of extremities  pain  outcome  
count                        16.0  16.0     16.0  
mean                          3.0   4.0      2.0  
std                           0.0   1.0      0.0  
min                           2.0   1.0      2.0  
25%                           3.0   3.0      2.0  
50%                           3.0   4.0      2.0  
75%                           3.0   5.0      2.0  
max                           4.0   5.0      2.0

## Итого по взрослым лошадям с без операции и погибших

### Нормальная температура взрослых лошадей 37,8 гр С

Средняя температура 38 гр С  
СКО 1 гр С  
Минимальная температура 36 гр С  
Максимальная температура 40 гр С  

С учетом того, что 75% также 38 гр С, то можно сделать заключение, что средняя температура не сильно превышает нормальную. 

### Нормальный пульс 30 - 40 уд/мин, но возможно и 20-30

Средний пульс 84 уд/мин  
СКО 24 уд/мин  
Минимальный пульс 42 уд/мин  
Максимальный пульс 120 уд/мин  

Средний пульс значительно превышает нормальное значение, начиная с первого квартиля также идет превышение значения

### Нормальная частота дыхания 8 - 10

Средняя частота дыхания 31  
СКО 9  
Минимальная частота дыхания 12  
Максимальная 48  

Среднее значение частоты дыхания почти в 3 раза превышает нормальное значение.  

### Боль

В основном лошади испытывали либо сильную постоянную, либо слабую или среднюю переодически боль. 

### Эвтаназия

In [344]:
data_info_ane = data_info_an[data_info_an['outcome'] == 3]
data_info_ane['pain'].value_counts()

4.0    6
2.0    3
3.0    3
1.0    1
Name: pain, dtype: int64

In [345]:
data_info_ane['rectal temperature'] = data_info_ane['rectal temperature'].fillna(data_info_ane['rectal temperature'].mean())
data_info_ane['respiratory rate'] = data_info_ane['respiratory rate'].fillna(data_info_ane['respiratory rate'].mode()[0])
data_info_ane['pain'] = data_info_ane['pain'].fillna(data_info_ane['pain'].mode()[0])
data_info_ane['pulse'] = data_info_ane['pulse'].fillna(data_info_ane['pulse'].median())
data_info_ane['temperature of extremities'] = data_info_ane['temperature of extremities'].fillna(data_info_ane['temperature of extremities'].median())
data_info_ane

/var/folders/xs/yywhsxs13dnc369zl2gzp53m0000gn/T/ipykernel_4514/3425652115.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_info_ane['rectal temperature'] = data_info_ane['rectal temperature'].fillna(data_info_ane['rectal temperature'].mean())
/var/folders/xs/yywhsxs13dnc369zl2gzp53m0000gn/T/ipykernel_4514/3425652115.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_info_ane['respiratory rate'] = data_info_ane['respiratory rate'].fillna(data_info_ane['respiratory rate'].mode()[0])
/var/folder

surgery?  Age  rectal temperature  pulse  respiratory rate  \
8         2.0    1           37.909091   80.0              36.0   
31        2.0    1           37.200000  108.0              12.0   
35        2.0    1           37.909091  104.0              24.0   
36        2.0    1           38.300000  112.0              16.0   
54        2.0    1           40.300000  114.0              36.0   
70        2.0    1           37.000000   60.0              20.0   
87        2.0    1           37.909091   40.0              16.0   
124       2.0    1           37.600000   48.0              20.0   
179       2.0    1           38.500000  100.0              20.0   
196       2.0    1           38.500000   60.0              20.0   
205       2.0    1           38.500000  100.0              20.0   
238       2.0    1           36.100000   88.0              20.0   
265       2.0    1           37.909091   56.0              68.0   
286       2.0    1           37.800000   82.0              12.0   
296       2.0    1           37.200000   72.0              24.0   

     temperature of extremities  pain  outcome  
8                           3.0   4.0      3.0  
31                          3.0   2.0      3.0  
35                          4.0   4.0      3.0  
36                          3.0   4.0      3.0  
54                          3.0   2.0      3.0  
70                          3.0   3.0      3.0  
87                          3.0   4.0      3.0  
124                         3.0   1.0      3.0  
179                         3.0   4.0      3.0  
196                         1.0   2.0      3.0  
205                         3.0   4.0      3.0  
238                         3.0   3.0      3.0  
265                         3.0   3.0      3.0  
286                         3.0   4.0      3.0  
296                         3.0   4.0      3.0

In [346]:
info_ane = data_info_ane.describe()
info_ane.round()

surgery?   Age  rectal temperature  pulse  respiratory rate  \
count      15.0  15.0                15.0   15.0              15.0   
mean        2.0   1.0                38.0   82.0              24.0   
std         0.0   0.0                 1.0   25.0              14.0   
min         2.0   1.0                36.0   40.0              12.0   
25%         2.0   1.0                37.0   60.0              18.0   
50%         2.0   1.0                38.0   82.0              20.0   
75%         2.0   1.0                38.0  102.0              24.0   
max         2.0   1.0                40.0  114.0              68.0   

       temperature of extremities  pain  outcome  
count                        15.0  15.0     15.0  
mean                          3.0   3.0      3.0  
std                           1.0   1.0      0.0  
min                           1.0   1.0      3.0  
25%                           3.0   2.0      3.0  
50%                           3.0   4.0      3.0  
75%                           3.0   4.0      3.0  
max                           4.0   4.0      3.0

## Итого по взрослым лошадям с без операции и эвтаназия

### Нормальная температура взрослых лошадей 37,8 гр С

Средняя температура 38 гр С  
СКО 1 гр С  
Минимальная температура 36 гр С  
Максимальная температура 40 гр С  

С учетом того, что 75% также 38 гр С, то можно сделать заключение, что средняя температура не сильно превышает нормальную. 

### Нормальный пульс 30 - 40 уд/мин, но возможно и 20-30

Средний пульс 82 уд/мин  
СКО 25 уд/мин  
Минимальный пульс 40 уд/мин  
Максимальный пульс 114 уд/мин  

Средний пульс значительно превышает нормальное значение, начиная с первого квартиля также идет превышение значения

### Нормальная частота дыхания 8 - 10

Средняя частота дыхания 24  
СКО 14  
Минимальная частота дыхания 12  
Максимальная 68  

Среднее значение частоты дыхания почти в 2,5 раза превышает нормальное значение.  

### Боль

В основном лошади испытывали чаще среднюю переодически боль. 

### Выжившие

In [347]:
data_info_anl = data_info_an[data_info_an['outcome'] == 1]
data_info_anl['pain'].value_counts()

1.0    25
3.0    20
2.0    16
5.0     1
4.0     1
Name: pain, dtype: int64

In [348]:
data_info_anl['rectal temperature'] = data_info_anl['rectal temperature'].fillna(data_info_anl['rectal temperature'].mean())
data_info_anl['respiratory rate'] = data_info_anl['respiratory rate'].fillna(data_info_anl['respiratory rate'].mode()[0])
data_info_anl['pain'] = data_info_anl['pain'].fillna(data_info_anl['pain'].mode()[0])
data_info_anl['pulse'] = data_info_anl['pulse'].fillna(data_info_anl['pulse'].median())
data_info_anl['temperature of extremities'] = data_info_anl['temperature of extremities'].fillna(data_info_anl['temperature of extremities'].median())
data_info_anl

/var/folders/xs/yywhsxs13dnc369zl2gzp53m0000gn/T/ipykernel_4514/1861360330.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_info_anl['rectal temperature'] = data_info_anl['rectal temperature'].fillna(data_info_anl['rectal temperature'].mean())
/var/folders/xs/yywhsxs13dnc369zl2gzp53m0000gn/T/ipykernel_4514/1861360330.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data_info_anl['respiratory rate'] = data_info_anl['respiratory rate'].fillna(data_info_anl['respiratory rate'].mode()[0])
/var/folder

surgery?  Age  rectal temperature  pulse  respiratory rate  \
2         2.0    1                38.3   40.0              24.0   
11        2.0    1                39.1   72.0              52.0   
17        2.0    1                37.5   48.0              24.0   
19        2.0    1                39.4  110.0              35.0   
21        2.0    1                38.4   48.0              16.0   
..        ...  ...                 ...    ...               ...   
278       2.0    1                37.8   40.0              16.0   
279       2.0    1                38.2   56.0              40.0   
284       2.0    1                38.5   54.0              16.0   
290       2.0    1                38.6   45.0              16.0   
294       2.0    1                38.5   40.0              16.0   

     temperature of extremities  pain  outcome  
2                           1.0   3.0      1.0  
11                          2.0   2.0      1.0  
17                          2.0   1.0      1.0  
19                          4.0   1.0      1.0  
21                          1.0   1.0      1.0  
..                          ...   ...      ...  
278                         1.0   1.0      1.0  
279                         4.0   2.0      1.0  
284                         1.0   3.0      1.0  
290                         2.0   1.0      1.0  
294                         1.0   2.0      1.0  

[73 rows x 8 columns]

In [349]:
info_anl = data_info_anl.describe()
info_anl.round()

surgery?   Age  rectal temperature  pulse  respiratory rate  \
count      73.0  73.0                73.0   73.0              73.0   
mean        2.0   1.0                38.0   54.0              24.0   
std         0.0   0.0                 1.0   15.0              10.0   
min         2.0   1.0                37.0   36.0               9.0   
25%         2.0   1.0                38.0   45.0              16.0   
50%         2.0   1.0                38.0   50.0              20.0   
75%         2.0   1.0                39.0   56.0              32.0   
max         2.0   1.0                40.0  112.0              52.0   

       temperature of extremities  pain  outcome  
count                        73.0  73.0     73.0  
mean                          2.0   2.0      1.0  
std                           1.0   1.0      0.0  
min                           1.0   1.0      1.0  
25%                           1.0   1.0      1.0  
50%                           2.0   2.0      1.0  
75%                           3.0   3.0      1.0  
max                           4.0   5.0      1.0

## Итого по взрослым лошадям с без операции и выжившие

### Нормальная температура взрослых лошадей 37,8 гр С

Средняя температура 38 гр С  
СКО 1 гр С  
Минимальная температура 37 гр С  
Максимальная температура 40 гр С  

С учетом того, что 75% также 38 гр С, то можно сделать заключение, что средняя температура не сильно превышает нормальную. 

### Нормальный пульс 30 - 40 уд/мин, но возможно и 20-30

Средний пульс 54 уд/мин  
СКО 15 уд/мин  
Минимальный пульс 36 уд/мин  
Максимальный пульс 112 уд/мин  

Средний пульс значительно превышает нормальное значение, начиная с первого квартиля также идет превышение значения

### Нормальная частота дыхания 8 - 10

Средняя частота дыхания 24  
СКО 10  
Минимальная частота дыхания 9  
Максимальная 52  

Среднее значение частоты дыхания почти в 2,5 раза превышает нормальное значение.  

### Боль

В основном лошади испытывали либо среднюю или слабую боль, либо вообще ее не испытывали. 